data/scripts/lib/quests.lua

```diff
@@ -0,0 +1,205 @@
+Quest = {}
+Quest.__index = Quest
+Quest.__list = {}
+Quest.__count = 0
+
+Mission = {}
+Mission.__index = Mission
+Mission.__list = {}
+Mission.__count = 0
+
+TrackedQuests = {}
+TrackedQuests.__index = TrackedQuests
+TrackedQuests.__list = {}
+
+function Quest.getById(id) return Quest.__list[id] end
+function Mission.getById(id) return Mission.__list[id] end
+function TrackedQuests.getByMissionsId(player, missionsId)
+	local playerId = player:getId()
+	local maxTrackedQuests = player:getMaxTrackedQuests()
+	TrackedQuests.__list[playerId] = {}
+	local trackeds = 0
+	local trackedQuests = TrackedQuests.__list[playerId]
+	for _, missionId in pairs(missionsId or {}) do
+		local mission = Mission(missionId)
+		if mission and mission:isStarted(player) then
+			trackeds = trackeds + 1
+			trackedQuests[mission] = Quest(mission.questId)
+			if trackeds >= maxTrackedQuests then
+				break
+			end
+		end
+	end
+	return trackedQuests, trackeds
+end
+
+setmetatable(Quest, {__call = function(_, id) return Quest.getById(id) end})
+setmetatable(Mission, {__call = function(_, id) return Mission.getById(id) end})
+setmetatable(TrackedQuests, {__call = function(_, player, missionsId) return TrackedQuests.getByMissionsId(player, missionsId) end})
```

Comment:
These belong to `Game` or `Player`, no need to create metatables. `Game.getQuestById(id)`, `Player.getTrackedQuests()` then `Quest.getMissionById(id)`